nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/listener/DocumentRoutingWorkflowInstancesCleanup.java

```diff
@@ -40,22 +41,20 @@
     public final static String CLEANUP_WORKFLOW_EVENT_NAME = "workflowInstancesCleanup";
 
     @Override
-    public void handleEvent(Event event) {
-        if (!CLEANUP_WORKFLOW_EVENT_NAME.equals(event.getName())
-                || Framework.isBooleanPropertyTrue(CLEANUP_WORKFLOW_INSTANCES_PROPERTY)) {
-            return;
-        }
-
-        int batchSize = Integer.parseInt(Framework.getProperty(CLEANUP_WORKFLOW_INSTANCES_BATCH_SIZE_PROPERTY, "100"));
-        DocumentRoutingService routing = Framework.getLocalService(DocumentRoutingService.class);
-        RepositoryManager repositoryManager = Framework.getLocalService(RepositoryManager.class);
+    public void handleEvent(EventBundle events) {
+        for (Event event : events) {
+            int batchSize = Integer.parseInt(
+                    Framework.getProperty(CLEANUP_WORKFLOW_INSTANCES_BATCH_SIZE_PROPERTY, "100"));
+            DocumentRoutingService routing = Framework.getLocalService(DocumentRoutingService.class);
+            RepositoryManager repositoryManager = Framework.getLocalService(RepositoryManager.class);
```

Comment:
These 3 can be moved out of the loop. The loop should just collect the repository names from the events, then do a second loop on the repository names collected (we don't want to do redundant work if there are several events, just in case).